docs/root/operations/admin.rst

```diff
@@ -444,6 +444,13 @@ modify different aspects of the server:
   Full-string matching can be specified with begin- and end-line anchors. (i.e.
   ``/stats?filter=^server.concurrency$``)
 
+  .. http:get:: /stats?histogram_buckets
```

Comment:
I have a draft PR #19546 which will hopefully be ready soon, where I introduce a 'type=' which makes me wonder about exactly how best to control this.

First -- is there any reason not to always be in histogram_buckets mode? Is there any reason you'd want to see the current mode?

The new PR introduces a '&type=' query-param which has possible values (Counters, Gauges, Histograms, TextReadouts, or All). I'm unsure whether this would best fit as a new category, or as a boolean property which only makes sense if you are displaying histograms.

Also, what are your thoughts about how this parameter works with format=json or format=prometheus? Or does this only make sense for text?

Reply:
> First -- is there any reason not to always be in histogram_buckets mode? Is there any reason you'd want to see the current mode?

I don't have enough experience on this to give a good opinion, but I didn't find a use for the current mode.

> The new PR introduces a '&type=' query-param which has possible values (Counters, Gauges, Histograms, TextReadouts, or All). I'm unsure whether this would best fit as a new category, or as a boolean property which only makes sense if you are displaying histograms.

I think a boolean property would suit this best.

> Also, what are your thoughts about how this parameter works with format=json or format=prometheus? Or does this only make sense for text?

From what I've understood, the current way format=prometheus displays histogram buckets cumulatively is the correct way for prometheus, so it seems like changing this would be unnecessary.

Using format=json outputs the same quantile summary data available from the plain text endpoint for histograms. If the quantile summary is replaced by the histogram_buckets output in the plain text endpoint, I think it should be replaced here too.

Reply:
So...can you add a test for the JSON mode as well then?

Reply:
I'll work on a JSON mode next. Also something that came to mind is that should there be a query parameter or an option of  seeing the original overlapping (or cumulative) histogram buckets? It should be easy to implement if there is any use for it.

Reply:
I think it's fine to have a query-param for the new mode. The absence of that query-params means you want the old mode. Am I missing something?

Reply:
I hope this explains what I meant:
`/stats` could output summary for histograms
`/stats?histogram_buckets` could output buckets for **cumulative** histograms
`/stats?histogram_buckets&nonoverlapping_buckets` could output buckets for **nonoverlapping** histograms

In addition, all versions would have a format=json output. Would this work with the changes to be introduced by #19546 and would this be useful?

Reply:
Yes, one query-param with 2 options sounds much better. I think something like this would work, although there might be a better word for nonoverlapping:
`/stats?histogram_buckets=cumulative`
`/stats?histogram_buckets=nonoverlapping`

Reply:
SGTM!

Reply:
oh...a better word for nonoverlapping might be "disjoint"

Reply:
Thank you! I'll start using this